src/mscorlib/src/System/UnSafeCharBuffer.cs

```diff
@@ -29,25 +29,36 @@ unsafe internal struct UnSafeCharBuffer{
             m_totalSize = bufferSize;    
             m_length = 0;
         }
-    
+
+        [System.Security.SecuritySafeCritical]  // auto-generated
+        public void AppendChar( char charToAppend ) {
+            if ( (m_totalSize - m_length) < 1 ) {
+                throw new IndexOutOfRangeException();
```

Comment:
We should probably look into changing this method (probably delete the class entirely) afterwards. In its current state this does not look like it will be inlined.